### PR TITLE
Restrict anonymous user capabilities

### PR DESCRIPTION
**Pull request by jefferai**

### CODE REVIEW

File: website/content/docs/concepts/security/permissions/assignable-permissions.mdx [hunk 1]
```diff
@@ -24,33 +24,51 @@ know.
...
 By granting the `no-op` action to users, they can see the resources in the
 output of a list command without needing other capability grants as well.
 
+### Anonymous User Restrictions
+
+Starting in Boundary 0.9.0, there are severe limits placed on the actions
+allowed to be assigned to the anonymous user `u_anon`. This is meant to avoid
+situations where an operator accidentally or mistakenly adds the anonymous user
+to a role that provides more privileges than might be intended for
+unauthanticated users.
```
Comment (jimlambrt) [inline website/content/docs/concepts/security/permissions/assignable-permissions.mdx:54]: spelling: unauthenticated

File: internal/perms/acl.go [hunk 2]
```diff
@@ -71,7 +71,7 @@ func NewACL(grants ...Grant) ACL {
 }
 
 // Allowed determines if the grants for an ACL allow an action for a resource.
-func (a ACL) Allowed(r Resource, aType action.Type) (results ACLResults) {
+func (a ACL) Allowed(r Resource, aType action.Type, isAnonUser bool) (results ACLResults) {
```
Comment (jimlambrt) [inline internal/perms/acl.go:74]: perhaps require the userId here (vs the bool) and then support an option like WithAllowAnon(bool) to support calls like perms.Parse(...) where you don't want to enforce the u_anon rules?

**Review by jimlambrt — CHANGES_REQUESTED** [review]
One spelling issue and one suggestion for consideration.

**Review by tmessi — COMMENTED** [review]
Looks good to me, but I second Jim's suggestion.

**Review by johanbrandhorst — DISMISSED** [review]

File: website/content/docs/concepts/security/permissions/assignable-permissions.mdx (hunk 1 above)
Comment (jefferai) [reply website/content/docs/concepts/security/permissions/assignable-permissions.mdx:54]: You know, it's weird -- I have the spell check plugin installed and sometimes it just....doesn't highlight things 😠

**Review by tmessi — DISMISSED** [review]

**Review by jimlambrt — DISMISSED** [review]

**Pull request merged**